Chatkit/Model/Room.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import CoreData
+
+public struct Room {
+    
+    // MARK: - Properties
+    
+    public let identifier: String
+    public let name: String?
+    public let isPrivate: Bool
+    public let creator: User
+    public let members: Set<User>?
+    public let typingMembers: Set<User>?
+    public let unreadCount: Int64
+    public let lastMessage: Message?
+    public let metadata: Metadata?
+    public let createdAt: Date
+    public let updatedAt: Date
+    public let deletedAt: Date?
+    
+    let objectID: NSManagedObjectID
+    
+    // MARK: - Initializers
+    
+    init(identifier: String, name: String?, isPrivate: Bool, creator: User, members: Set<User>?, typingMembers: Set<User>?, unreadCount: Int64, lastMessage: Message?, metadata: Metadata?, createdAt: Date, updatedAt: Date, deletedAt: Date?, objectID: NSManagedObjectID) {
+        self.identifier = identifier
+        self.name = name
+        self.isPrivate = isPrivate
+        self.creator = creator
+        self.members = members
+        self.typingMembers = typingMembers
+        self.unreadCount = unreadCount
+        self.lastMessage = lastMessage
+        self.metadata = metadata
+        self.createdAt = createdAt
+        self.updatedAt = updatedAt
+        self.deletedAt = deletedAt
+        self.objectID = objectID
+    }
+    
+}
+
+// MARK: - Hashable
+
+extension Room: Hashable {
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(self.identifier)
+    }
+    
+}
+
+// MARK: - Equatable
+
+extension Room: Equatable {
+    
+    public static func == (lhs: Room, rhs: Room) -> Bool {
+        // Metadata is intentionally excluded from this comparison.
+        return lhs.identifier == rhs.identifier
```

Comment:
If metadata is excluded; can we guarantee then that two objects will truly be equal then if we don't compare all properties?

Reply:
No we cannot. The problem is that `Metadata` is defined as `[String : Any]` and `Any` does not implement `Equatable` in Swift.

We could try to use the following approach to solve this problem: https://stackoverflow.com/questions/47632263/comparing-two-string-any-dictionaries-in-swift-4?rq=1